doc/release-notes/release-notes-4.1.0.md

```diff
@@ -0,0 +1,358 @@
+Notable changes
+===============
+
+Migration to Clang and static libc++
+------------------------------------
+
+`zcashd` now builds its C++ (and C) dependencies entirely with a pinned version
+of Clang, and statically links libc++ instead of dynamically linking libstdc++.
+This migration enables us to reliably use newer C++ features while supporting
+older LTS platforms, be more confident in the compiler's optimisations, and
+leverage security features such as sanitisers and efficient fuzzing.
+
+Additionally, because both Clang and rustc use LLVM as their backend, we can
+optimise across the FFI boundary between them. This reduces the cost of moving
+between C++ and Rust, making it easier to build more functionality in Rust
+(though not making it costless, as we still need to work within the constraints
+of the C ABI).
```

Comment:
```suggestion
leverage security features such as sanitisers and efficient fuzzing. In future,
this will also allow optimizing across the boundary between Rust and C++.
```